integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/NessieUtils.scala

```diff
@@ -167,29 +169,49 @@ object NessieUtils {
       currentCatalog: CatalogPlugin,
       catalog: Option[String]
   ): NessieApiV1 = {
+    val maybeIcebergCatalog = getBaseIcebergCatalog(currentCatalog, catalog)
+    val errorPre =
+      "The command works only when the catalog is a NessieCatalog or a RESTCatalog using the Nessie Catalog Server"
+    val errorPost =
+      "Either set the catalog via USE <catalog_name> or provide the catalog during execution: <command> IN <catalog_name>."
+    require(maybeIcebergCatalog.isDefined, errorPre + ". " + errorPost)
+
+    val icebergCatalog = maybeIcebergCatalog.get
+
     val catalogName = catalog.getOrElse(currentCatalog.name)
-    val sparkConf = SparkSession.active.sparkContext.conf
-    val catalogConf = sparkConf
-      .getAllWithPrefix(s"spark.sql.catalog.$catalogName.")
-      .toMap
 
-    val catalogImpl = catalogConf.get("catalog-impl")
-    val catalogErrorDetail = catalogImpl match {
-      case Some(clazz) => s"but $catalogName is a $clazz"
-      case None =>
-        s"but spark.sql.catalog.$catalogName.catalog-impl is not set"
-    }
-    // Referring to https://github.com/apache/iceberg/blob/master/nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java
-    // Not using fully-qualified class name to provide protection from shading activities (if any)
-    require(
-      catalogImpl
-        .exists(impl => impl.endsWith(".NessieCatalog")),
-      s"The command works only when the catalog is a NessieCatalog ($catalogErrorDetail). Either set the catalog via USE <catalog_name> or provide the catalog during execution: <command> IN <catalog_name>."
-    )
+    val nessieClientConfigMapper: java.util.function.Function[String, String] =
+      icebergCatalog.getClass.getSimpleName match {
+        case "NessieCatalog" =>
+          val sparkConf = SparkSession.active.sparkContext.conf
+          val catalogConf = sparkConf
+            .getAllWithPrefix(s"spark.sql.catalog.$catalogName.")
+            .toMap
+          x => catalogConf.getOrElse(x.replace("nessie.", ""), null)
+        case "RESTCatalog" =>
+          val catalogProperties = getCatalogProperties(icebergCatalog)
```

Comment:
Instead of inspecting the catalog's property map, can't we do like above and simply invoke `sparkConf.getAllWithPrefix(s"spark.sql.catalog.$catalogName.")`?

Reply:
These are _not_ Spark properties - these are Iceberg properties

Reply:
It works from Spark config "down" to Iceberg catalog config, but not the other way around.